Vagrantfile

```diff
@@ -56,11 +56,45 @@ def vm_memory
   $vb_memory.nil? ? $vm_memory : $vb_memory
 end
 
+PERSISTENT_DISK_DIR = File.join(File.expand_path('~'), '.bricks-data')
+PERSISTENT_DISK = File.join(PERSISTENT_DISK_DIR, 'persistent_data.vdi')
+MACHINE_ID_FILE= File.join(File.expand_path('~'),'.bricks/clusters/coreos-vagrant/.vagrant/machines/core-01/virtualbox/id')
+
+def machine_id
+    File.exists?(MACHINE_ID_FILE)? File.read(MACHINE_ID_FILE) : ""
+end
+
+
 def vm_cpus
   $vb_cpus.nil? ? $vm_cpus : $vb_cpus
 end
 
 Vagrant.configure("2") do |config|
+
+  if File.exist?(MACHINE_ID_FILE)
+    config.trigger.before :destroy do |trigger|
+      trigger.info = "dettach virtual disk"
```

Comment:
typo: dettach -> detach

Reply:
This is actually detaching, the flag `--medium none` actual removes the medium in that slot. See: https://www.virtualbox.org/manual/ch08.html#vboxmanage-storageattach